python/lsst/meas/algorithms/brightStarStamps.py

```diff
@@ -0,0 +1,280 @@
+        self._checkRadius(innerRadius, "INNER_RADIUS")
+        self._checkRadius(outerRadius, "OUTER_RADIUS")
+        return None
+
+    def _checkRadius(self, radiusValue, metadataName):
+        """Ensure provided annulus radius is consistent with that present
+        in metadata. If metadata does not contain annulus radius, add it.
+        """
+        # if a radius value is already present in metadata, ensure it matches
+        # the one given
+        if self._metadata.exists(metadataName):
+            if radiusValue is not None:
+                if self._metadata.get(name=metadataName) != radiusValue:
+                    raise AttributeError("BrightStarStamps instance already contains different annulus radii "
+                                         + f"values ({metadataName}).")
+        # if not already in metadata, a value must be provided
+        elif radiusValue is None:
+            raise AttributeError("No radius value provided for the AnnularFlux measurement "
+                                 + f"({metadataName}).")
+        else:
+            self._metadata.add(name=metadataName, value=radiusValue)
+        return None
+
+    def writeFits(self, filename):
+        """Write a single FITS file containing all bright star stamps.
+        """
+        # ensure metadata contains current number of objects
+        self._metadata.setInt(name="N_STARS", value=len(self))
+
+        # add full list of Gaia magnitudes, IDs and annularFlxes to shared
+        # metadata
+        self._metadata.set(name="G_MAGS", value=self.getMagnitudes())
+        self._metadata.setString(name="GAIA_IDS", value=self.getGaiaIds())
+        self._metadata.set(name="ANNULAR_FLUXES", value=self.getAnnularFluxes())
+
+        # create primary HDU with global metadata
+        fitsPrimary = afw.fits.Fits(filename, "w")
+        fitsPrimary.createEmpty()
+        fitsPrimary.writeMetadata(self._metadata)
+        fitsPrimary.closeFile()
+
+        # add all stamps and mask planes
+        for stamp in self.getMaskedImages():
+            stamp.getImage().writeFits(filename, mode='a')
+            stamp.getMask().writeFits(filename, mode='a')
+        return None
+
+    @classmethod
+    def readFitsWithOptions(cls, filename, options):
+        """Read bright star stamps from FITS file.
+
+        Returns
+        -------
+        bss : `BrightStarStamps`
+            Collection of bright star stamps.
+        """
+        # extract necessary info from metadata
+        visitMetadata = afw.fits.readMetadata(filename, hdu=0)
+        nbStarStamps = visitMetadata.getInt("N_STARS")
+        innerRadius = visitMetadata.get("INNER_RADIUS")
+        outerRadius = visitMetadata.get("OUTER_RADIUS")
+        gaiaGMags = visitMetadata.getArray("G_MAGS")
+        gaiaIds = visitMetadata.getArray("GAIA_IDS")
+        annularFluxes = visitMetadata.getArray("ANNULAR_FLUXES")
+        # check if a bbox was provided
+        kwargs = {}
```

Comment:
Why are you creating a dict for one value?

Reply:
I can't rightly say... Fixed.

Reply:
Ah! Yes I can, though admittedly there's probably a much better way to go about this.

It is so that `kwargs` can be passed along to `ImReader` and `maskReader` below (lines 273-274), regardless of whether a bbox was specified. If not, it's an empty dictionary.